src/components/CommonComponents/Pagination/utils/buildPaginationModel.ts

```diff
@@ -0,0 +1,138 @@
+export interface PageType {
+  className?: string;
+  type: string;
+  num: number;
+  disabled?: boolean;
+  selected?: boolean;
+}
+
+export function buildPaginationModel(
```

Comment:
This sounds a little bit overly complex, no?

I'm quite sure you can accomplish a fairly simple pagination logic without all this super extent logic 🤔

Reply:
I might be wrong, of course, as I see you're doing full pagination with numbers. But there might be already packages out there, eg, in the NPM registry that are simpler so we don't need to maintain this complex logic by ourselves 😄